Sources/AppcuesKit/Data/Analytics/AnalyticsTracker.swift

```diff
@@ -141,23 +141,21 @@ extension Activity {
     }
 
     mutating func append(_ activity: Activity) {
-        // the only thing we support merging here is additional events.
-        // if the user or group, or any associated user or group properties are updated,
-        // those cause any pending activity to be flushed, then the user or group
-        // activity immediately flushed as an individual item - there is no concept
-        // of merging content across user or group updates.  The reason is that the
-        // activity prior to that update needs to be associated to the correct user/group
-        // and the activity after that update needs to be associated with the new user/group
+        // this will merge any additional events or profile updates.
+        // we do not support any update to the user ID or group ID here, as a change
+        // in those values would trigger an immediate flush of pending items then the new update
+        // sent. The reason is that the activity prior to that update needs to be associated to
+        // the previous user/group and the activity after that update needs to be associated with
+        // the new user/group
         if let newEvents = activity.events {
             let existingEvents = events ?? []
             events = existingEvents + newEvents
+        }
 
-            // additional events can also cause autoproperty updates, merge those in
-            newEvents.forEach {
-                if let autoProps = $0.autoProperties {
-                    profileUpdate = (profileUpdate ?? [:]).merging(autoProps) { _, new in new }
-                }
-            }
+        // merge in any updated auto props from events or other pending profile updates
+        if let newProfileUpdate = activity.profileUpdate {
+            let existingProfileUpdate = profileUpdate ?? [:]
+            profileUpdate = existingProfileUpdate.merging(newProfileUpdate) { _, new in new }
```

Comment:
I believe this to actually be safe, rewriting some of the previous comments above, and matching Android more closely.  As long as we are careful to never allow the actual user ID to change in the middle of a batch, we can combine the properties into a merged set.

Reply:
Any worries about queuing the profile update impacting future qualifications or personalizations?

Reply:
I don't think so, since any qualification eligible analytics would cause an immediate flush of anything in the queue as well along with it - so it should always have the most up to date profile information at qualification time (or personalization on a qualified subsequent flow). But, if you think I'm missing something there please let me know.

Reply:
Right. I was overlooking the fact the we'd need a flushing event. I guess technically an app could `Appcues.show()` and then have stale personalization data, but not worried about that.